spec/models/provider_spec.rb

```diff
@@ -747,49 +747,20 @@
     end
   end
 
-  describe "#visa_sponsorship_publishable?" do
-    context "when year is 2022 and provider has declared sponsorship" do
-      let(:provider) do
-        build(
-          :provider,
-          can_sponsor_student_visa: true,
-          can_sponsor_skilled_worker_visa: false,
-          recruitment_cycle: build(:recruitment_cycle, year: 2022),
-        )
-      end
+  describe "#sponsors_visa?" do
+    context "provider has declared sponsorship" do
+      let(:provider) { build(:provider, can_sponsor_student_visa: true, can_sponsor_skilled_worker_visa: false) }
 
       it "returns true" do
-        expect(subject.visa_sponsorship_publishable?).to be(true)
+        expect(subject.sponsors_visa?).to be(true)
       end
     end
 
-    context "when year is 2021 and provider has not declared sponsorship" do
-      let(:provider) do
-        build(
-          :provider,
-          can_sponsor_student_visa: nil,
-          can_sponsor_skilled_worker_visa: false,
-          recruitment_cycle: build(:recruitment_cycle, year: 2021),
-        )
-      end
+    context "provider has not declared sponsorship" do
+      let(:provider) { build(:provider, can_sponsor_student_visa: nil, can_sponsor_skilled_worker_visa: false) }
 
       it "returns true" do
-        expect(subject.visa_sponsorship_publishable?).to be(true)
-      end
-    end
-
-    context "when year is 2022 and provider has not declared sponsorship" do
-      let(:provider) do
-        build(
-          :provider,
-          can_sponsor_student_visa: nil,
-          can_sponsor_skilled_worker_visa: false,
-          recruitment_cycle: build(:recruitment_cycle, year: 2022),
-        )
-      end
-
-      it "returns false" do
-        expect(provider.visa_sponsorship_publishable?).to be(false)
+        expect(subject.sponsors_visa?).to be(false)
```

Comment:
:face_with_head_bandage:

Reply:
I just realised from working on the Publish frontend that both visa attributes need values. Will fix.